### ISSUE TITLE
Include associated ethevents in each utxo returned by /account.get_utxos

### ISSUE THREAD

**Issue by unnawut** [omisego/elixir-omg#1154]
Written with @okalouti:

One user story for the block explorer is to visualize UTXOs by address, including their status - for example, deposited, received, exit started, exit finalized, exit challenged.

To enable that, we would need to call `account.get_utxos` from the informational API. Currently, `/account.get_utxos` returns the following.

```javascript
{
  "data": [
    {
      "amount": 499999999999992200,
      "blknum": 92000,
      "currency": "0x0000000000000000000000000000000000000000",
      "oindex": 1,
      "owner": "0xe4c8c1f49fe031f879278853a01634004022f2eb",
      "txindex": 0,
      "utxo_pos": 92000000000001
    }
  ],
}
```

As clarified with Jacob, to implement this user story we would need the call to return an array of ethevents with associated information (i.e. timestamps) and references (i.e. ETH or Plasma txhash)

See designs here: https://gallery.io/projects/MCHbtQVoQ2HCZSli4_VqE738/files/MCGj8QigksOSzzEV7R9wwiRo

Example:

```javascript
{
  "data": [
    {
      "amount": 499999999999992200,
      "blknum": 92000,
      "currency": "0x0000000000000000000000000000000000000000",
      "oindex": 1,
      "owner": "0xe4c8c1f49fe031f879278853a01634004022f2eb",
      "txindex": 0,
      "utxo_pos": 92000000000001,
      "events": [
        {
          "type": "deposit",
          "timestamp": 1540300000,
          "eth_hash": "0x10918e623557ab5310017372421f9a92bedb716212b660c33bec330befc14e67",
          "status": "confirmed"
        },
        {
          "type": "transfer",
          "timestamp": 1540400000,
          "plasma_hash": "0x0017372421f9a92bedb7163557ab5310befc14e67212b660c33bec3310918e62",
          "status": "pending"
        }
      ]
    }
  ],
}
```

Possible events:

```javascript
{
  "type": "deposit",
  "timestamp": 1540300000,
  "eth_hash": "0x10918e623557ab5310017372421f9a92bedb716212b660c33bec330befc14e67",
  "status": "confirmed"
},
{
  "type": "transfer",
  "timestamp": 1540400000,
  "plasma_hash": "0x0017372421f9a92bedb7163557ab5310befc14e67212b660c33bec3310918e62",
  "status": "pending"
},
{
  "type": "startStandardExit",
  "timestamp": 1540500000,
  "eth_hash": "0x14e67212b660c3bedb7163310918e623557ab5310befcbec30017372421f9a92",
  "status": "confirmed"
},
{
  "type": "processExit",
  "timestamp": 1540600000,
  "eth_hash": "0x310918e623557ab00173befc14e67212b660c33bec72421f9a92bedb71635310",
  "status": "pending"
},
{
  "type": "startInFlightExit",
  "timestamp": 1540700000,
  "eth_hash": "0xb716353103310918e623557ab00172b660c33bec72421f9a92bedbefc14e6721",
  "status": "confirmed"
},
{
  "type": "exitChallenged",
  "timestamp": 1540700000,
  "eth_hash": "0xab00172b66b716353103310918e6235570c33bec72421f9a9befc14e67212bed",
  "status": "confirmed"
}
```

**Comment by unnawut** [comment]
Also open to discussion if this belongs in Info API

**Comment by pdobacz** [comment]
I'm supportive. Note that the relations between utxos and eth events can be very different. Are we're aiming for this to read, that `events` are just events that "somehow" relate to the UTXO in question, and it is on the client's (using the data contained in `events`) to figure out what the relation is.

I think this is a sound approach, but I want to make this explicit and put under scrutiny.

**Comment by unnawut** [comment]
> Are we're aiming for this to read, that events are just events that "somehow" relate to the UTXO in question, and it is on the client's (using the data contained in events) to figure out what the relation is.

Yeah hence the `"type": "deposit"`, `"type": "transfer"`, etc. Better just provide the raw data.

...

Although I must highlight 2 more things for scrutiny:

1. Notice that the deposit event has `eth_hash` but transfer event has `plasma_hash`. Might be something to improve but we havn't got a better idea yet.

2. Also I'm not sure if the `"status"` field for each event makes sense. Since those events should already be confirmed by all the thresholds we set on `EthereumEventListener`?

**Comment by jrhite** [comment]
Assuming you mean plasma transaction by 'transfer event', bear in mind a
plasma transaction is the only entity in watcher with no actual
corresponding with eth txn hash. Blocks and UTXOs both do.

**Comment by pdobacz** [comment]
> Also I'm not sure if the "status" field for each event makes sense. Since those events should already be confirmed by all the thresholds we set on EthereumEventListener?

yeah, today all would be `confirmed`, but we could have no-confirmation listeners too.

Ah, what you may like is to give the Ethereum height of the event (with current listeners that would be approximate! deep reorgs can change it, but that's a different can of worms). Then whoever is interested in finality of the event in blocks can figure it out.

**Comment by unnawut** [comment]
> Assuming you mean plasma transaction by 'transfer event', bear in mind a plasma transaction is the only entity in watcher with no actual corresponding with eth txn hash. Blocks and UTXOs both do.

and

> I'd say, let's not mix plasma transactions with these events. If we'd like to reference an event with a plasma transaction, then maybe that should be a BlockSubmitted event instead? It could then list two events - block where the UTXO was created, and then spent (if we decide to show spent utxos in some form of this view).

> OTOH, if you want richer information about the UTXO, creating_transaction and spending_transaction can be shown outside the events in some form? 🤔 All depends on the actual use case of the API call.

Right BlockSubmitted makes more sense in that events array, instead of a plasma transaction.

And yes since one utxo belongs up to 2 transfers (in and out), maybe `creating_txhash` and `spending_txhash` outside `events` array should be enough. We already have those DB fields in `txoutputs` too actually.
